_components/form-controls/10-file-input.md

```diff
@@ -0,0 +1,54 @@
+---
+title: File input
+parent: Form controls
+order: 10
+lead: File inputs allow users to attach a file
+---
+
+{% include code/preview.html component="file-input" %}
+{% include code/accordion.html component="file-input" %}
+
+<div class="usa-accordion usa-accordion--bordered site-accordion-docs">
+  <button class="usa-button-unstyled usa-accordion__button"
+      aria-expanded="true" aria-controls="file-input-docs">
+    Guidance
+  </button>
+  <div id="file-input-docs" aria-hidden="false" class="usa-accordion__content site-component-usage">
+
+    <h4>When to use the file input</h4>
+    <ul class="usa-content-list">
+      <li><strong>Documents are required.</strong> You should only ask users to upload files when it is critical to service delivery.</li>
+    </ul>
+    <h4>When to consider something else</h4>
+    <ul class="usa-content-list">
+      <li><strong>Documents are optional.</strong> Don’t ask users to provide documents if you do not require them.</li>
+      <li><strong>Upload functionality.</strong> The USWDS file input does not currently upload files.</li>
+    </ul>
+    <h4>Usability guidance</h4>
+    <ul class="usa-content-list">
+      <li><strong>Allow multiple file formats.</strong> Users won’t have the same software. Allow multiple types of files so user won’t need to download new software.</li>
```

Comment:
I'm not clear on this one. I get that we don't want to be overly restrictive in what file formats can be attached. I just don't understand the point about users not needing to download new software.

Reply:
I'm wrestling with how to word this. You want to be flexible in what formats to allow. Like, don't require that files be `.docx`. Some people might use pages, which by default gives a `.pages` extension